pkgs/development/python-modules/htmllistparse/default.nix

```
@@ -0,0 +1,20 @@
+{ lib, stdenv, buildPythonPackage, fetchPypi, beautifulsoup4, html5lib, requests, fusepy }:
+
+buildPythonPackage rec {
+  pname = "htmllistparse";
+  version = "0.6.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-bcimvwPIQ7nTJYQ6JqI1GnlbVzzZKiybgnFiEBnGQII=";
+  };
+
+  propagatedBuildInputs = [ beautifulsoup4 html5lib requests fusepy ];
+
+  meta = with lib; {
+    homepage = "https://github.com/gumblex/htmllisting-parser";
+    description = "Python parser for Apache/nginx-style HTML directory listing";
+    license = [ licenses.mit ] ;
+    maintainers = with maintainers; [ hexchen ];
```

```suggestion
    maintainers = with maintainers; [ hexchen ];
    mainProgram = "htmllistparse";
```

The package should list the main program because the old behaviour of assuming the pname and mainProgram is the same is deprecated.

---

The package does not have an executable. It is a python library.

---

Well, technically it does. The executable is rehttpfs, which is a separate tool from the python library packaged with the library. I do not understand why the maintainer did that, but it doesn't make much sense to explicitly name it as the mainProgram when this package is used primarily as a library.

---

I see sorry for the inconvenience (: